### PR TITLE
Fix ability to edit courses via the admin and `canvas_id` readonly logic, improve validation (#1509)

### PR DESCRIPTION
**Pull request by ssciolla**
This PR fixes two bugs introduced by my work on PR #1468. It tries to do the following:

- Make courses editable again via the Django Admin interface by modifying the `clean` method in `CourseForm` to only validate `canvas_id` on course creation (`canvas_id` was always left out when editing since it was read only, which lead to errors being thrown).
- Fix a side-effect of the previous change where when a course was edited, subsequent creation of new courses still lableed `canvas_id` as read only. This was fixed by using the `get_readonly_fields` method rather than updating the instance variable.
- Try to improve the validation by disallowing zero and negative Canvas IDs. This wasn't in response to any bugs created by the previous change, but instead exploration of edge cases it makes sense to handle.

Test cases
- [ ] Create a new course with a valid course ID. Subsequently editing the course is possible (i.e. with the start and end times), but `canvas_id` is read only.
- [ ] Try to create a new course with the following invalid values. All should fail with appropriate error messages.
  - [ ] a decimal
  - [ ] a negative integer and/or decimal
  - [ ] zero
  - [ ] no entry
- [ ] Edit an existing course (maybe created automatically by a launch), and then try to create a new course. You should be able to edit `canvas_id` without issue.

If you think of more cases to test, please let me know.

### CODE REVIEW

**Comment by pushyamig** [comment]
I will review this today. Thanks for the Fix @ssciolla

File: dashboard/admin.py [hunk 1]
```diff
@@ -94,9 +103,11 @@ def _courseviewoption(self, obj):
...
         return format_html('<a href="{}">Link</a>', obj.absolute_url)
 
-    def change_view(self, request, object_id, form_url='', extra_content=None):
-        self.readonly_fields = self.readonly_fields + ('canvas_id',)
-        return super(CourseAdmin, self).change_view(request, object_id)
+    def get_readonly_fields(self, request: HttpRequest, obj: Any | None = ...) -> tuple[str, ...]:
+        readonly_fields: tuple[str, ...] = self.readonly_fields
+        if obj is not None:
+            readonly_fields += ('canvas_id',)
+        return readonly_fields
```
Comment (pushyamig) [inline dashboard/admin.py:110]: This logics work. basically the `obj` contains `canvas_id`. can we make check if the obj contains 'canvas_id` or value of the canvas_id rather than just checking for None?

Comment (ssciolla) [reply dashboard/admin.py:110]: I'll verify, but I think `obj` is the model instance. I think it's `None` if we're creating a new object.

Comment (ssciolla) [reply dashboard/admin.py:110]: Verified. `obj` is `None` on create, `Course` when modifying.

Comment (pushyamig) [reply dashboard/admin.py:110]: Ok .

**Review by pushyamig — APPROVED** [review]
LGTM! Thanks

**Comment by ssciolla** [comment]
Thanks, @pushyamig!
